analysis/priestdiscipline/src/modules/problems/ramps/analyzer.ts

```diff
@@ -0,0 +1,155 @@
+import SPELLS from 'common/SPELLS';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import CASTS_THAT_ARENT_CASTS from 'parser/core/CASTS_THAT_ARENT_CASTS';
+import Events, { CastEvent } from 'parser/core/Events';
+import { Options } from 'parser/core/Module';
+import EventHistory from 'parser/shared/modules/EventHistory';
+import StatTracker from 'parser/shared/modules/StatTracker';
+
+import GlobalCooldown from '../../core/GlobalCooldown';
+import Atonement from '../../spells/Atonement';
+const POWER_WORD_RADIANCE_CAST_TIME_MS = 2000;
+const ALLOWED_PRE_EVANG = [
+  SPELLS.POWER_WORD_RADIANCE.id,
+  SPELLS.POWER_WORD_SHIELD.id,
+  SPELLS.SHADOW_MEND.id,
+  SPELLS.RAPTURE.id,
+  SPELLS.SHADOWFIEND.id,
+  SPELLS.EVANGELISM_TALENT.id,
+  SPELLS.SHADOW_WORD_PAIN.id,
+];
+const RAMP_STARTERS = [
+  SPELLS.SHADOW_WORD_PAIN.id,
+  SPELLS.PURGE_THE_WICKED_TALENT.id,
+  SPELLS.SHADOW_MEND.id,
+  SPELLS.RAPTURE.id,
+  SPELLS.POWER_WORD_SHIELD.id,
+];
+
+export type ProblemHash = {
+  problemDescription: string;
+  index: number;
+  problemType: string;
+  timestamp: string;
+  gcdTime?: number;
+};
+
+type EvangelismRampType = {
+  rampInfo: CastEvent[];
+  evangelismTimestamp: string;
+};
```

Comment:
Brief comment description on what these are and what the fields mean may be helpful